### PR TITLE
Adding pytim

### PR DESCRIPTION
**Pull request by Marcello-Sega**

### CODE REVIEW

File: mdakits/pytim/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+## make sure that it is appropriate for how you run tests.
+#test_dependencies:
+#  - pip install mdtraj "pytest==8.1.1" "codecov==2.1.13" "pytest-cov==5.0.0" 
+
+## str: the organisation name the MDAKit falls under
+project_org: Marcello-Sega
+
+## str: the development status of the MDAKit
+## See https://pypi.org/classifiers/ for development status classifiers.
+development_status: Production/Stable
```
Comment (orbeckst) [inline mdakits/pytim/metadata.yaml:90]: Pleas fix: The correct classifier is `Development Status :: 5 - Production/Stable`

File: mdakits/pytim/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+## Otherwise you need to include commands to make the tests available. 
+## For example, if the tests are in the repository at the top level under `./tests`:
+## First use `git clone latest` to either clone the top commit for "development code" checks or check out
+## the latest tag for "latest release" checks. Then then run pytest:
+##    - git clone latest
+##    - pytest -v ./tests
+## Feel free to ask for advice on your pull request!
+run_tests:
+  - git clone latest
+  - pip install "mdtraj" "codecov==2.1.13" "pytest-cov==5.0.0" "cython>=0.24.1"  "scipy>=1.6.0" "gsd<=3.2.1" "PyWavelets>=0.5.2" "scikit-image>=0.14.2" "sphinx>=1.4.3" "matplotlib" "dask>=1.1.1"
```
Comment (orbeckst) [inline mdakits/pytim/metadata.yaml:75]: I think it would be cleaner if the `pip install` were under `test_dependencies` or did that not work?

Comment (orbeckst) [reply mdakits/pytim/metadata.yaml:75]: Are mdtraj, gsd, PyWavelets, scikit-image optional dependencies that are not automatically installed with `pip install pytim`?

Also, is the restriction on `gsd<=3.2.1` necessary? (MDAnalysis itself requires `gsd>3.0.0` so we should be able to install a version that works for both packages but having less restrictive version requirements would be better for future compatibility.)

Comment (orbeckst) [reply mdakits/pytim/metadata.yaml:75]: This `pip install` replaces numpy 2.0.1 with numpy 1.26 – probably because of one of the dependencies. pytim itself has `numpy>=1.16.0` in https://github.com/Marcello-Sega/pytim/blob/master/requirements.txt . Is this an problem @IAlibay ?

**Review by orbeckst — CHANGES_REQUESTED** [review]
Thank you for submitting pytim. Looks very good.

I have to required, minor fixes (see comments) and one question where I need some input from @IAlibay about changes in the installed environment.

File: mdakits/pytim/metadata.yaml (hunk 2 above)
Comment (Marcello-Sega) [reply mdakits/pytim/metadata.yaml:75]: I'll try to clean up this part. I'm also updating the dependencies in another minor update of pytim, removing python2.8 support and requiring MDAnalysis>=2.7 & numpy>=1.22.3. All is good with the GitHub workflows, but CircleCI (both on GitHub and on coda-forge) is giving me a headache – some tests are returning different numbers and it's not about truncation, it really looks like some methods are behaving differently, but I can't find out which ones, or what is different betwee GitHub workflows and CircleCI.

Bottom line, it might take a bit, sorrry :(

Comment (Marcello-Sega) [reply mdakits/pytim/metadata.yaml:75]: About numpy 2.0.x, I see that `mdanalysis 2.7.0 requires numpy<2.0,>=1.22.3,`

Comment (Marcello-Sega) [reply mdakits/pytim/metadata.yaml]: Re. numpy 2, more specifically:

```
pip install --user --upgrade "numpy>2.0.0" "MDAnalysis>=2.7.0"

Collecting numpy>2.0.0
  Using cached numpy-2.0.1-cp39-cp39-macosx_11_0_arm64.whl.metadata (60 kB)
Requirement already satisfied: MDAnalysis>=2.7.0 in /Users/marcello/Library/Python/3.9/lib/python/site-packages (2.7.0)
INFO: pip is looking at multiple versions of mdanalysis to determine which version is compatible with other requirements. This could take a while.
Collecting MDAnalysis>=2.7.0
  Using cached MDAnalysis-2.7.0-cp39-cp39-macosx_11_0_arm64.whl.metadata (46 kB)
ERROR: Cannot install mdanalysis==2.7.0 and numpy>2.0.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested numpy>2.0.0
    mdanalysis 2.7.0 depends on numpy<2.0 and >=1.22.3
    The user requested numpy>2.0.0
    mdanalysis 2.7.0 depends on numpy<2.0 and >=1.22.3

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip to attempt to solve the dependency conflict
```

Comment (orbeckst) [reply mdakits/pytim/metadata.yaml]: You're right @Marcello-Sega — I had at the back of my head "we're supporting numpy 2" ... but that's in develop not in 2.7.0.

**Review by orbeckst — APPROVED** [review]
All looks good to me, thank you for making the effort to register pytim. I am approving the registration. 🎉
